File: bindings/php/examples/sign257.php
<?php

declare(strict_types=1);

/**
 * crypto_sign257 (DSTU 4145 m=257, T-199/T-204): generate a signing keypair, sign a message,
 * verify it.
 *
 * Run: php -d extension=<path to dstu_core_php.dll/.so> examples/sign257.php
 */

$signingKey = dstu_core_sign257_keygen();
$verifyingKey = dstu_core_sign257_verifying_key($signingKey);

$message = 'a message whose origin and integrity matter';
$signature = dstu_core_sign257_message($signingKey, $message);
if (!dstu_core_sign257_verify($verifyingKey, $message, $signature)) {
    throw new RuntimeException('verification failed');
}

printf("signed and verified a %d-byte message\n", strlen($message));

if (!dstu_core_sign257_verify($verifyingKey, 'a different message', $signature)) {
    echo "signature over a different message correctly rejected\n";
}

File: bindings/ruby/examples/sign257.rb
# frozen_string_literal: true

# crypto_sign257 (DSTU 4145 m=257, T-199/T-204): generate a signing keypair, sign a message,
# verify it.
#
# Run: ruby examples/sign257.rb

$LOAD_PATH.unshift(File.expand_path("../lib", __dir__))
require "dstu_core"

signing_key = DstuCore.sign257_keygen
verifying_key = DstuCore.sign257_verifying_key(signing_key)

message = "a message whose origin and integrity matter"
signature = DstuCore.sign257_message(signing_key, message)
raise "verification failed" unless DstuCore.sign257_verify(verifying_key, message, signature)

puts "signed and verified a #{message.bytesize}-byte message"

unless DstuCore.sign257_verify(verifying_key, "a different message", signature)
  puts "signature over a different message correctly rejected"
end
